runner/browserrunner.ts

```diff
@@ -0,0 +1,211 @@
+/**
+ * @license
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+import * as _ from 'lodash';
+import * as chalk from 'chalk';
+import * as cleankill from 'cleankill';
+import * as wd from 'wd';
+import {Config} from './config';
+
+export interface Stats {
+  status: string;
+  passing?: number;
+  pending?: number;
+  failing?: number;
+}
+
+interface NodeCB<T> {
+  (err: any, value?: T): void;
+}
+
+export interface BrowserDef extends wd.Capabilities {
+  id: number;
+  url: string;
+  sessionId: string;
+  deviceName?: string;
+}
+
+// Browser abstraction, responsible for spinning up a browser instance via wd.js and
+// executing runner.html test files passed in options.files
+export class BrowserRunner {
+  timeout: number;
+  browser: wd.Browser;
+  stats: Stats;
+  sessionId: string;
+  timeoutId: NodeJS.Timer;
+
+  constructor(public emitter: NodeJS.EventEmitter, public def: BrowserDef, public options: Config, public doneCallback: NodeCB<BrowserRunner>) {
```

Comment:
I assume that adding `public` to the parameter declares it as a class field. That's... neat :) but I wouldn't mix with the long hand class field declarations.